2015/proposals/li-yanghao.md

```diff
@@ -0,0 +1,83 @@
+
+1. **Accessing the resolution services.** This could be built by using [pytaxize](https://github.com/sckott/pytaxize) to resolve species names with specific parameters. Current `pytaxize` supports resolving names by some services, such as [Global Names Resolver](http://resolver.globalnames.org/). We could also build out `pytaxize` to support more resolution services and have the Retriever use it.
+
+2. **Determining the election strategy.** Since each resolution service may return many candidate names, we need to choose a best name according to the returned scores of services and some qualities of the candidate names. We also need to determine a few default standard data sources since it would be much faster than searching all data sources. For example, there are [181 sources](http://resolver.globalnames.org/data_sources) in `GNR`. My current idea is simple, which is to choose the name with highest average score from all data sources and the score also must be higher than a threshold. I will continue to talk with mentors to determine the final election strategy.
+
+3. **Updating the user interfaces.** This project also needs to update the user interface to provide options interface for users before downloading a dataset. This interface contains some options, such as whether to use name resolution and choices of data sources.
+
+## Schedule of Deliverables
+
+###Before May 25th
+
+Get more familiar with the Retriever codes.
+Determine the election strategy.
+
+### May 25th -  June 7th
+
+Add functions to access more name resolution services in pytaxize.
+
+### June 8th - June 21th
+
+Finish adding functions in pytaxize and test them.
+Write the module in Retriever to query names through pytaxize ports.
+
+### June 22th - July 5th
+
+Do mid-term evaluations.
+Finish up the module to query names and process responses to choose the best name.
+
+### July 6th - July 19th
+
+Test on some datasets to evaluate the replacement results and make some updates of the election strategy.
+
+### July 20th - August 2rd
+
+Update the user interfaces for users to choose some options about the name resolution.
+
+### August 3rd - August 16th
+
+Finish the user interfaces and make sure it works well with back-end codes.
+Test the entire code.
+
+### August 17th - August 21th 19:00 UTC
+
+Week to scrub code, write tests, improve documentation, etc
+
+## Future works
+
+Continue to work on Retriever Project and finish some requests or fix some bugs in Retriever Issues.
+Improve pytaxize by adding more functions.
+
+## Open Source Development Experience
+
+I'm sorry, I don't have any open source development experience before. However, I'm always looking forward to have a chance to contribute.  
```

Comment:
First, you don't need to apologize!

Second, looking at your GitHub profile it seems you do have some repositories and contributions, and you've engaged with the Retriever team as well. Can you think of any contributions you've made that you're proud of or that might be relevant here, or lessons you've learned working in groups?